python/ray/workflow/common.py

```diff
@@ -318,11 +370,32 @@ def data(self) -> WorkflowData:
             del self._prepare_inputs
         return self._data
 
+    @property
+    def ref(self) -> Optional[WorkflowStaticRef]:
+        return self._ref
+
+    @classmethod
+    def from_ref(cls, workflow_ref: WorkflowStaticRef) -> "Workflow":
+        inputs = WorkflowInputs(args=None, workflows=[], workflow_refs=[])
+        data = WorkflowData(
+            func_body=None,
+            inputs=inputs,
+            name=None,
+            step_options=WorkflowStepRuntimeOptions.make(
+                step_type=StepType.FUNCTION),
+            user_metadata={})
+        wf = Workflow(data)
+        wf._ref = workflow_ref
+        return wf
+
     def __reduce__(self):
-        raise ValueError(
-            "Workflow[T] objects are not serializable. "
-            "This means they cannot be passed or returned from Ray "
-            "remote, or stored in Ray objects.")
+        if self._ref is None:
+            raise ValueError(
+                "Workflow[T] objects are not serializable, except it is a "
+                "reference to a workflow (when workflow.ref is not 'None'). "
```

Comment:
Please revert this error message. It is user-facing, and should not contain implementation details.